draft-ietf-tls-ctls.md

```diff
@@ -237,36 +237,54 @@ defined in {{RFC8446, Section 8.4}}.
 
 #### `dh_group`
 
-Value: a single `NamedGroup` ({{!RFC8446, Section 4.2.7}}) to use for key establishment.
+Value: a single `CTLSKeyShareGroup` to use for key establishment.
 
-This is equivalent to a literal "supported_groups" extension
-consisting solely of this group.
+~~~~
+struct {
+    NamedGroup group_name;
+    uint16 key_share_length;
```

Comment:
Indeed it is somewhat redundant with the group name. The goal is that two implementation don't have diverging interpretation of what "self-delimiting" means, and serialize messages differently.

Why is this important? The security of TLS rely on the non-ambiguity of the transcript hash: the serialization of two transcripts are equal if and only if the transcripts are equal.
Since cTLS modify the transcript serialization by compressing it, we need to check that this property still holds in cTLS.
If two implementations don't agree on the transcript serialization, they will have a different interpretation of what the transcript hash means, and we can't prove any security properties between their interaction when executing their (diverging) implementations of cTLS: an active attacker could interact with the two participants such that their transcript hashes are equal but correspond to different TLS 1.3 transcripts and break confidentiality / authenticity.
(such an attack would probably be difficult to implement in practice, but we can't prove it doesn't happen)

Adding an explicit `length` field like this is the easiest way to proceed with analysis tools, but adding a column to the Supported Groups registry would also be fine.